src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -225,7 +225,19 @@ object LoadVCF {
     val infoSignature = headerSignature(infoHeader)
 
     val formatHeader = header.getFormatHeaderLines
-    val genotypeSignature: Type = if (reader.genericGenotypes) headerSignature(formatHeader).getOrElse(TStruct.empty) else TGenotype
+    val genotypeSignature: Type =
+      if (reader.genericGenotypes) {
+        val sig = headerSignature(formatHeader).getOrElse(TStruct.empty)
+        val callFields: Set[String] = reader.asInstanceOf[GenericRecordReader].callFields
+
+        callFields.foldLeft(sig) { (s, n) => s.fieldOption(n) match {
```

Comment:
I'd probably pass in callFields to headerSignature and building the right type the first time rather than modifying it after the fact.